tests/prague/eip7692_eof_v1/eip7069_extcall/test_calls.py

```diff
@@ -470,3 +471,214 @@ def test_eof_calls_legacy_mstore(
+    )
+
+    calling_storage = {
+        slot_code_worked: value_code_worked,
+        slot_call_result: value_eof_call_reverted
+        if destination_opcode == Op.REVERT
+        or (opcode == Op.EXTDELEGATECALL and not destination_is_eof)
+        else value_eof_call_failed,
+    }
+
+    post = {
+        calling_contract_address: Account(storage=calling_storage),
+        destination_contract_address: Account(storage={}),
+    }
+
+    state_test(
+        env=env,
+        pre=pre,
+        post=post,
+        tx=tx,
+    )
+
+
+@pytest.mark.parametrize(
+    ["opcode", "suffix"],
+    [
+        [Op.EXTCALL, [0, 0, 0]],
+        [Op.EXTDELEGATECALL, [0, 0]],
+    ],
+    ids=["extcall", "extdelegatecall"],
+)
+@pytest.mark.parametrize("fail_opcode", [Op.REVERT(0, 0), Op.INVALID], ids=["revert", "invalid"])
+def test_eof_calls_eof_then_fails(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    sender: EOA,
+    opcode: Op,
+    suffix: list[int],
+    fail_opcode: Op,
+):
+    """Test EOF contracts calling EOF contracts and failing after the call"""
+    env = Environment()
+    destination_contract_address = pre.deploy_contract(contract_eof_sstore)
+
+    caller_contract = Container.Code(
+        Op.SSTORE(slot_call_result, opcode(destination_contract_address, *suffix))
+        + Op.SSTORE(slot_code_worked, value_code_worked)
+        + fail_opcode,
+    )
+    calling_contract_address = pre.deploy_contract(caller_contract)
+
+    tx = Transaction(
+        sender=sender,
+        to=Address(calling_contract_address),
+        gas_limit=50000000,
+        gas_price=10,
+        protected=False,
+        data="",
+    )
+
+    post = {
+        calling_contract_address: Account(storage=Storage()),
+        destination_contract_address: Account(storage=Storage()),
+    }
+
+    state_test(
+        env=env,
+        pre=pre,
+        post=post,
+        tx=tx,
+    )
+
+
+@pytest.mark.parametrize(
+    ["opcode", "suffix"],
+    [
+        [Op.EXTCALL, [0, 0, 0]],
+        [Op.EXTDELEGATECALL, [0, 0]],
+        [Op.EXTSTATICCALL, [0, 0]],
+    ],
+    ids=["extcall", "extdelegatecall", "extstaticall"],
+)
+@pytest.mark.parametrize(
+    "target_account_type",
+    (
+        "empty",
+        "EOA",
+        "LegacyContract",
+        "EOFContract",
+    ),
+    ids=lambda x: x,
+)
+def test_eof_calls_clear_return_buffer(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    sender: EOA,
+    opcode: Op,
+    suffix: list[int],
+    target_account_type: str,
+):
+    """Test EOF contracts calling clears returndata buffer"""
+    env = Environment()
+    filling_contract_code = Container.Code(
+        Op.MSTORE8(0, int.from_bytes(value_returndata_magic, "big")) + Op.RETURN(0, 32),
+    )
+    filling_callee_address = pre.deploy_contract(filling_contract_code)
+
+    match target_account_type:
+        case "empty":
+            target_address = b"\x78" * 20
+        case "EOA":
+            target_address = pre.fund_eoa(10**18)
```

Comment:
```suggestion
            target_address = pre.fund_eoa()
```
Default should make this test more portable.